### PR TITLE
[SYCLomatic] Fix return type of begin/end_recording

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by tangjj11 — APPROVED** [review]
lgtm

**Comment by yihwang-nv** [comment]
Do we need to keep a stable and unified function signature? Avoid to following the compiler version changes?

File: clang/runtime/dpct-rt/include/dpct/graph.hpp [hunk 1]
```diff
@@ -84,11 +119,17 @@ static inline bool begin_recording(sycl::queue *queue_ptr) {
 /// were recorded.
 /// \param [out] graph A pointer to a command_graph_ptr pointer where the
 /// command graph will be assigned.
+#if (__SYCL_COMPILER_VERSION && __SYCL_COMPILER_VERSION < 20240427)
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/graph.hpp]: is it works if always return void?

Comment (daiyaan-ahmed6) [reply clang/runtime/dpct-rt/include/dpct/graph.hpp]: Done.

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**